osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/SetMbusUserKeyByChannelRequestMessageProcessor.java

```diff
@@ -34,11 +38,8 @@ public SetMbusUserKeyByChannelRequestMessageProcessor() {
 
     @Override
     protected Serializable handleMessage(final DlmsConnectionManager conn, final DlmsDevice device,
-            final Serializable requestObject) throws OsgpException {
-
-        this.assertRequestObjectType(SetMbusUserKeyByChannelRequestDataDto.class, requestObject);
-
+            final RequestWithMetadata<SetMbusUserKeyByChannelRequestDataDto> request) throws OsgpException {
         return this.configurationService.setMbusUserKeyByChannel(conn, device,
-                (SetMbusUserKeyByChannelRequestDataDto) requestObject);
+                CorrelatedObject.from(request, request.getRequestObject()));
```

Comment:
A call to `CorrelatedObject.from(request)` seems nicer, as it takes both metadata and object from `RequestWithMetadata` and appears to be made for this type of situation.